playbooks/prerun-network-checks.yml

```diff
@@ -1,133 +1,188 @@
 
-- name: prerequisite checks
+- name: Run preliminary network checks for search hosts
   hosts: search
   become: yes
   roles:
     - role: '../roles/helper_modules'
   tasks:
-  - name: check repo connection
+  - name: Check repo connection
     include_tasks: "../roles/helper_modules/tasks/check_port.yml"
     vars:
       checked_host: "{% if groups.repository | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.repository[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
       checked_port: "{{ ports_cfg.repository.http }}"
       delegate_target: "{{ groups.repository | first }}"
 
-- name: prerequisite checks
+- name: Run preliminary network checks for transformers hosts
   hosts: transformers
   become: yes
   roles:
     - role: '../roles/helper_modules'
   tasks:
-  - name: check activemq connection
+  - name: Check activemq connection
     include_tasks: "../roles/helper_modules/tasks/check_port.yml"
     vars:
       checked_host: "{% if groups.activemq | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.activemq[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
       checked_port: "{{ ports_cfg.activemq[activemq_protocol] }}"
       delegate_target: "{{ groups.activemq | first }}"
+    when: groups.activemq | default([]) | length > 0
 
-- name: prerequisite checks
+- name: Run preliminary network checks for syncservice hosts
   hosts: syncservice
   become: yes
   roles:
     - role: '../roles/helper_modules'
   tasks:
-  - name: check repo connection
+  - name: Check db connection
+    include_tasks: "../roles/helper_modules/tasks/check_port.yml"
+    vars:
+      checked_host: "{% if groups.database | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.database[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
+      checked_port: "{{ ports_cfg.postgres.sql }}"
+      delegate_target: "{{ groups.database | first }}"
+    when: repo_db_url == ""
+
+  - name: Check repo connection
     include_tasks: "../roles/helper_modules/tasks/check_port.yml"
     vars:
       checked_host: "{% if groups.repository | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.repository[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
       checked_port: "{{ ports_cfg.repository.http }}"
       delegate_target: "{{ groups.repository | first }}"
-  - name: check activemq connection
+
+  - name: Check activemq connection
     include_tasks: "../roles/helper_modules/tasks/check_port.yml"
     vars:
       checked_host: "{% if groups.activemq | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.activemq[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
       checked_port: "{{ ports_cfg.activemq[activemq_protocol] }}"
       delegate_target: "{{ groups.activemq | first }}"
+    when: groups.activemq | default([]) | length > 0
 
-- name: prerequisite checks
+- name: Run preliminary network checks for acc hosts
+  hosts: acc
+  become: yes
+  roles:
+    - role: '../roles/helper_modules'
```

Comment:
Why are you calling the role if you include the tasks explicitly?
To be fair, that stuff should not even be a role. If you feel like moving it to the playbook level feel free to do so.
If not, then just remove the `roles` execution

Reply:
All the other play of this playbook do that, I simply duplicated the ADW play and changed all `adw` references with `acc`, since that seems to be what was required for ACC as well. Isn't that to load the library `listen_port.py` that is then used inside the `check_port.yml` task? If that isn't the reason, I do not know, do did it like that for other plays so I just did the same.

Reply:
Oh lord yes you're most likely right. That's a really ugly hack :-/ 
I've done some quick tests and it seems a single role is needed for the complete set of plays in this playbook to find the module file.
That said, I have created a new ticket to completely get rid of that roles: OPSEXP-2147 (shadows #438)
For now we can move forward with current code in this PR.